### ISSUE TITLE
Provide some samples or documentation

### ISSUE THREAD

**Issue by grumpyf0x48** [jbangdev/jbang#363]
Apart from ```jbang help``` and the examples of jbang use given in the readme, it would be great to have:

1. A documentation - Picocli generated ?
2. A cheat sheet usable with https://github.com/cheat/cheat

The second one is very efficient, as you document only the most frequent usages of a command, so you find very quickly the command you need.

**Comment by maxandersen** [comment]
Agreed. Would you be interested taking a stab on making any these two happen ?

**Comment by maxandersen** [comment]
Btw. By documentation in #1 I assume you refer to man pages or similar ?

**Comment by grumpyf0x48** [comment]
> Agreed. Would you be interested taking a stab on making any these two happen ?

Yes, i'm interested on starting a cheat sheet.

The idea i have is

- create a new file jbang in a new folder cheat of the jbang repository
- when the commands and flags of jbang will not change too much (maybe after the 1.0 release), then submit this cheat to the community ones (https://github.com/cheat/cheatsheets)

**Comment by maxandersen** [comment]
Sounds like a good start. 

The command structure is fairly stable at this stage so I wouldn't be too concerned submitting to cheat Repo sooner rather than later. 

But let's get it started first and see where it goes.

**Comment by grumpyf0x48** [comment]
> Btw. By documentation in #1 I assume you refer to man pages or similar ?

Yes man page is an option for Linux users but maybe not for Windows ones.

I was more thinking of the abilility to generate documentation from Picocli annotations, it could help to keep documentation up to date.

**Comment by maxandersen** [comment]
There is https://picocli.info/#_generate_man_page_documentation but I haven't explored how it works and if worth it. 

PR's/suggestions welcome as man page for jbang would make sense imo.

**Comment by grumpyf0x48** [comment]
> There is https://picocli.info/#_generate_man_page_documentation but I haven't explored how it works and if worth it.
> 
> PR's/suggestions welcome as man page for jbang would make sense imo.

Will take a look at that next week.

**Comment by maxandersen** [comment]
just saw https://cli.github.com/manual/ - that is a sweet way of documenting a cli. wonder if we could style the docs generated with similar navigation/style ?

**Comment by abelsromero** [comment]
> wonder if we could style the docs generated with similar navigation/style ?

If you are just talking about aggregating docs with a lateral menu. Antora could be an option.
The https://github.com/asciidoctor/asciidoctor.js/ uses it to generate the manual https://asciidoctor-docs.netlify.app/asciidoctor.js/ while keeping the docs in the same repo.

**Comment by grumpyf0x48** [comment]
manpage generated by picocli-codegen are added to a manpage folder in distribution.zip.

After build, you can see what it looks like with:

```console
man build/docs/manpage/jbang.1 
```

i would have liked to zip each file before inclusion in distribution.zip but it looks like i'm too new to Gradle for that yet :-(
